lib/course_planner/tasks.ex

```diff
@@ -47,22 +47,32 @@ defmodule CoursePlanner.Tasks do
     end
   end
 
-  def get_user_tasks(id) do
-    Repo.all(from t in Task, where: t.user_id == ^id, preload: [:user])
+  def get_for_user(id, now) do
+    Repo.all(from t in Task, where: t.user_id == ^id and t.finish_time > ^now, preload: [:user])
   end
 
-  def get_unassigned_tasks do
-    Repo.all(from t in Task, where: is_nil(t.user_id))
+  def get_unassigned(now) do
+    Repo.all(from t in Task, where: is_nil(t.user_id) and t.finish_time > ^now)
   end
 
-  def grab(task_id, user_id) do
+  def grab(task_id, user_id, now) do
     case get(task_id) do
       {:ok, task} ->
         task
         |> Task.changeset()
+        |> validate_finish_time(now)
```

Comment:
it would be good to validate also if the task was already assigned to someone else, let's say a volunteer leaves the task index page open the whole day then they decide to click the link to grab a task, this task may be assigned already in the meantime